Tests/AppTests/QueryPerformanceTests.swift

```diff
@@ -129,13 +137,15 @@ class QueryPerformanceTests: XCTestCase {
               r.stars,
               r.last_activity_at,
               r.summary,
-              v.package_name
+              v.package_name,
+              pr.type AS product_type
             FROM packages p
               JOIN repositories r ON r.package_id = p.id
               JOIN versions v ON v.package_id = p.id
+              LEFT JOIN products pr ON pr.version_id = v.id
             WHERE v.reference ->> 'branch' = r.default_branch
             """)
-        try await assertQueryPerformance(query, expectedCost: 9_800, variation: 200)
+        try await assertQueryPerformance(query, expectedCost: 13_700, variation: 200)
```

Comment:
That's a significant increase but total time is still ~100ms so nothing to worry about given that we only run it once every ~20s. I'm just glad that we've got visibility into this.

Reply:
Details for reference:

```
ℹ️ TEST:        test_Search_refresh()
ℹ️ COST:        13682.03
ℹ️ EXPECTED:    9800.0 ± 200.0
ℹ️ ACTUAL TIME: 98.161ms
/__w/SwiftPackageIndex-Server/SwiftPackageIndex-Server/Tests/AppTests/QueryPerformanceTests.swift:148: error: QueryPerformanceTests.test_Search_refresh : failed - Total cost of 13682.03 above threshold of 10000.0 (incl variation)

Query plan:

Nested Loop  (cost=9640.49..13682.03 rows=478 width=319) (actual time=53.248..98.161 rows=6864 loops=1)
  Join Filter: (r.package_id = p.id)
  ->  Hash Right Join  (cost=9640.21..13530.45 rows=478 width=271) (actual time=53.103..84.361 rows=6864 loops=1)
        Hash Cond: (pr.version_id = v.id)
        ->  Seq Scan on products pr  (cost=0.00..3526.97 rows=95597 width=58) (actual time=0.012..12.838 rows=95692 loops=1)
        ->  Hash  (cost=9635.52..9635.52 rows=375 width=245) (actual time=53.048..53.051 rows=4691 loops=1)
              Buckets: 8192 (originally 1024)  Batches: 1 (originally 1)  Memory Usage: 1361kB
              ->  Hash Join  (cost=875.27..9635.52 rows=375 width=245) (actual time=6.537..49.707 rows=4691 loops=1)
                    Hash Cond: ((v.package_id = r.package_id) AND ((v.reference ->> 'branch'::text) = r.default_branch))
                    ->  Seq Scan on versions v  (cost=0.00..8356.73 rows=75073 width=195) (actual time=0.014..32.561 rows=75074 loops=1)
                    ->  Hash  (cost=804.91..804.91 rows=4691 width=208) (actual time=6.452..6.453 rows=4691 loops=1)
                          Buckets: 8192  Batches: 1  Memory Usage: 1187kB
                          ->  Seq Scan on repositories r  (cost=0.00..804.91 rows=4691 width=208) (actual time=0.010..4.016 rows=4691 loops=1)
  ->  Index Scan using packages_pkey on packages p  (cost=0.28..0.30 rows=1 width=80) (actual time=0.002..0.002 rows=1 loops=6864)
        Index Cond: (id = v.package_id)
Planning Time: 3.769 ms
Execution Time: 98.576 ms
```